File: atmega8l/ReflowRegion.cpp
/*
 * Andy's Workshop Reflow Controller ATMega328p firmware
 * Copyright (c) 2015 Andy Brown. http://www.andybrown.me.uk
 * Please see website for licensing terms.
 */

#include "Application.h"


namespace awreflow {

  /*
   * Reflow regions for the leaded profile
   */

  const ReflowRegion LeadReflowRegions[6] PROGMEM = {
    { 90,150  },      // preheat
    { 180,180 },      // soak
    { 200,210 },      // ramp up
    { 220,210 },      // reflow
    { 240,180 },      // ramp down
    { 270,25  }       // cool down
  };


  /*
   * Reflow regions for the lead-free profile
   */

  const ReflowRegion LeadFreeReflowRegions[6] PROGMEM = {
    { 90,150  },
    { 180,217 },
    { 200,240 },
    { 220,240 },
    { 240,217 },
    { 270,25  },
  };
}

File: android/awreflow/src/main/java/uk/me/andybrown/awreflow2/ReflowRegion.java
// Andy's Workshop Open Source Reflow Controller
// Copyright (c) 2015 Andy Brown. All rights Reserved.
// Please see website (http://www.andybrown.me.uk) for full license details.

package uk.me.andybrown.awreflow2;

/*
 * The reflow region class encapsulates a region on the profile, for example
 * "preheat", "soak" etc.
 */

public class ReflowRegion {

  protected String _name;               // name of the reflow region
  protected double _endTime;            // ending time in seconds
  protected double _endTemperature;     // ending temperature


  /**
   * Constructor
   * @param name The name of the region
   * @param etime The ending time
   * @param etemp The ending temp
   */

  public ReflowRegion(String name,double etime,double etemp) {
    _name=name;
    _endTime=etime;
    _endTemperature=etemp;
  }


  /*
   * Getters
   */

  public String getName() { return _name; }
  public double getEndTemperature() { return _endTemperature; }
  public double getEndTime() {  return _endTime; }
}
